ast/compile.go

```diff
@@ -4762,7 +4762,18 @@ func checkUnusedAssignedAndArgVars(loc *Location, stack *localDeclaredVars, used
 	unused = unused.Diff(rewrittenUsed)
 
 	for _, gv := range unused.Sorted() {
-		errs = append(errs, NewError(CompileErr, loc, "assigned var %v unused", dvs.reverse[gv]))
+		switch dvs.occurrence[dvs.reverse[gv]] {
+		case assignedVar:
+			for i := range body {
+				if body[i].Vars(VarVisitorParams{}).Contains(gv) {
```

Comment:
Traversing each statement in the body, to then walk each to find the var _is_ expensive. But as I think we've discussed before, since this only happens when compilation is aborted with an error, I think this is acceptable.

We could, however, break out of iterating the body once the unused var is found.

Reply:
fair, I've added a `break` here